### ISSUE TITLE
Normal prompt behaviour for machine context / windows authentication?

### ISSUE THREAD

**Issue by davros1973** [MohammadYounes/OWIN-MixedAuth#87]
Hi - I'm embarrassed by my ignorance by asking this - I'm only 6-weeks into programming as a job for the first time, and like-wise ASP.NET.  I've spent the entire weekend experimenting and reading but I'm just getting more confused.

Workgroup (not AD).

--------
Question: OWIN-MixedAuth - MVC project... sample download on Vis Studio 2015 ... when running on (IIS) localhost or machine-name hostname I can press the Windows log-in button and it logs-in without a prompt, displaying my username in the menu area, but any other (IIS) hostname binding variation or attempting from a browser on a different machine in the Workgroup results in a browser windows authentication prompt to achieve the same thing.  Is that normal?  The problem is, I don't know what is normal!
--------
nb: When prompted, after the initial prompt/successful log-in I can log-out and back-in without a further prompt leading me to assume the cookie authentication works fine.

So, that is,  running on IIS with a non-localhost or machine name hostname, on a workgroup network (not AD) ... so set to Machine context, not domain.  (Using SQL Server sa account for now and everyone-NTFS permission etc. - most privilege everywhere as I try to work out what's going on, and tried with NTLM and Negotiate types just while I learn (trying each possibility individually against various other settings)- only left Windows Authentication on the local path as per the sample - with parent override set - "allow" and "read/write" etc. Various app-pool settings - custom or built-in or pass-through variations.  Release & Debug builds - run on IIS from Vis Studio 2015, and also, published "release" to a separate IIS folder/site, tried with various bindings, internal DNS and/or hosts file set and/or over SSL while I tried variations / experiments and on different browsers with various intranet-site and trusted site settings) etc. etc.   Brute-force experimentation. I assumed I was doing something "wrong".

I'd assumed I'd never normally be prompted but maybe that's normal behaviour?  I don't have an AD at home yet to test the domain case but I had similar behaviours on Friday at work on a domain machine but using the older MVC5-mixedauth sample integrated with my first .Net application (been a steep learning curve) leading me to spend all my weekend trying to understand it.  Unsuccessfully.

nb: regardless of this "issue" I'm having (probably an ignorance issue more than anything) - I appreciate this work ... contrary to current appearances it has itself helped me learn more quickly about ASP.NET authentication.

Windows 10, 1703, IIS 10, Chrome, Firefox, Edge, IE

**Comment by davros1973** [comment]
Update (briefly for once) - at work, on the domain.  When I change, in IIS, the hostname binding to anything but localhost or the machine host name then, on the SAME machine as the IIS server, I still get prompted.  BUT ... on the machine next door to me (still logged on as me though), and just adding the new arbitrary IIS hostname to the hosts file on that machine for now (pending requesting a DNS entry), then things work without prompts.  I'm not so bothered about the machine-context (non-domain) issue any longer which maybe I conflated in my ignorance with the domain environment - I'm satisfied the behaviour I desire in the domain is realisable.  (Crossed fingers).

**Comment by MohammadYounes** [comment]
Hi - thanks for all the details. 

The prompt depends on **Security Settings/User Authentication** of the current site zone:

![auto login](https://cloud.githubusercontent.com/assets/4712046/6040904/b1c35c44-ac7f-11e4-9526-7086d71c058b.png)

The other thing to note is Browsers will cache the windows credentials and use them for subsequent windows authentication requests, **No prompt will be displayed** asking for user credentials (Even if your internet settings were set to prompt for username and password). You need to either close the browser or invalidate the browser credentials cache.

Nevertheless, The log off link will still be required to log the user out. For example, a user logs in using windows identity then decides to switch to another local user account.

I hope this clarifies the cases where prompt is shown or not.

Thanks.

**Comment by davros1973** [comment]
Hi Mohammad - thanks for the great software and for taking the time to make that comment ... 

I did try changing those settings repeatedly along with other other combinations ... but I just realised I never tried restarting the computer (VM's) ... although that was the original setting I think when I added sites to the "intranet zone".  I'll try once more, with a restart, when I get a chance at home. And if it still fails, I'll record what I did precisely to maybe illuminate anything silly I'm doing.  

I might have to add another "issue" though I'll read through all the existing again first ... but I've had a problem also at home, and at work, with SSL - where the log-on via windows authentication gets "stuck" and requires two attempts - but I'll do that when I'm more sure of myself.

Thanks again. :)
